aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -70,6 +70,12 @@ Tensor sum_batching_rule(const Tensor& self, IntArrayRef dims, bool keepdim, opt
   return self_physical.getPhysicalToLogicalMap().apply(result);
 }
 
+Tensor sin_batching_rule(const Tensor& self, optional<ScalarType> dtype) {
```

Comment:
batching-related changes probably don't have to go in this PR, unless it breaks some existing behavior

Reply:
This changes is necessary as 
https://github.com/pytorch/pytorch/blob/5fc60f02df02240fadaeba05ca17752655fb6728/aten/src/ATen/BatchingRegistrations.cpp#L1074-L1075

Requires the unary op with signature `Tensor (*)(const Tensor&)`.
But this PR updates `sin`'s signatuer to `Tensor (*)(const Tensor&, c10::optional<ScalarType>)`

Reply:
fyi @zou3519

Reply:
Looks good to me, thanks for updating it @kshitij12345